server/backend/api/app/integration/forest_client/forest_client.py

```diff
@@ -0,0 +1,55 @@
+import logging
+from http import HTTPStatus
+
+import requests
+from api.config import config
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ForestClientService():
+    """
+    The class is used for making requests to get information from Forest Client API.
+    Api is located at BC API Service Portal: https://api.gov.bc.ca/devportal/api-directory/3179.
+    API-key needs to be requested from the portal.
+    Spec of API:
+        test: https://nr-forest-client-api-test.api.gov.bc.ca/
+        prod: https://nr-forest-client-api-prod.api.gov.bc.ca/
+    """
+    def __init__(self):
+        self.api_base_url = config.get_forest_client_api_baseurl()
+        self.api_clients_url = f"{self.api_base_url}/api/clients"
+        self.API_TOKEN = config.get_forest_client_api_token()
+        self.headers = {"Accept": "application/json", "X-API-KEY": self.API_TOKEN}
+
+        # See Python: https://requests.readthedocs.io/en/latest/user/advanced/
+        self.session = requests.Session()
+        self.session.headers.update(self.headers)
+
+    def find_by_client_number(self, p_client_number: str):
+        """
+        Find Forest Client(s) information based on p_client_number search query field.
+
+        :param p_client_number: Forest Client Number string (8 digits).
+                                Note! Current Forest Client API can only do exact match.
+                                '/api/clients/findByClientNumber/{clientNumber}'
+        :return: Search result as List for a Forest Client information object.
+                 Current Forest Client API returns exact one result or http status
+                 other than 200 with message content. The intent for FAM search is for
+                 wild card search and Forest Client API could be capable of doing that
+                 in next version.
+        """
+        url = f"{self.api_clients_url}/findByClientNumber/{p_client_number}"
+        LOGGER.debug(f"ForestClientService find_by_client_number() - url: {url}")
+        r = self.session.get(url)
```

Comment:
Two error scenarios:
1. HTTP get fails entirely (no service running) - likely a python error like TCP connection timeout
2. The call returns HTTP status code 5xx indicating error in forest client.

Reply:
Thanks Basil! As discussed, will try to see what happen in python and deal with 5xx error.

Reply:
Python doc says: ```In the event of a network problem (e.g. DNS failure, refused connection, etc), Requests will raise a [ConnectionError](https://requests.readthedocs.io/en/latest/api/#requests.ConnectionError) exception``` so same as normal behavior, framework will handle that.
However, python best practice is we need to set "timeout" limit as there is no default one from python.

Reply:
Error handling for 400/500 and network error were addressed and discussed yesterday.
A note from handling 404 from ForestClient API as it uses this code to indicate return of "Client Not Found" so we handle this case in try-catch block and return empty for our search purpose.